lib/spack/spack/binary_distribution.py

```diff
@@ -36,22 +38,297 @@
 from spack.stage import Stage
 from spack.util.gpg import Gpg
 
+
+#: default root, relative to the Spack install path
+default_manager_root = os.path.join(spack.paths.opt_path, 'spack')
+
 _build_cache_relative_path = 'build_cache'
 
-BUILD_CACHE_INDEX_TEMPLATE = '''
-<html>
-<head>
-  <title>{title}</title>
-</head>
-<body>
-<ul>
-{path_list}
-</ul>
-</body>
-</html>
-'''
 
-BUILD_CACHE_INDEX_ENTRY_TEMPLATE = '  <li><a href="{path}">{path}</a></li>'
+class BinaryDistributionCacheManager(object):
+    """
+    The BinaryDistributionCacheManager stores and manages both:
+
+    1. cached buildcache index files (the index.json describing the built
+    specs available on a mirror)
+
+    2. a cache of all the conrcrete built specs available on all the
+    configured mirrors.
+    """
+
+    def __init__(self, cache_root=None):
+        self._cache_root = default_manager_root
```

Comment:
Recommend lines 60-62:

```suggestion
        self._cache_root = cache_root or default_manager_root
```